### PR TITLE
macOS: Allow to sign with ECDSA token

### PR DESCRIPTION
**Pull request by metsma**
Signed-off-by: Raul Metsma <raul@metsma.ee>

### CODE REVIEW

**Comment by MrJoe** [comment]
Can't merge the PR as it is now. There are 2 commits which both have exactly the same git commit message (except for the prefix) "Allow to sign with ECDSA token" and it also includes unrelated cmakefile changes.
We want to keep our keep commit message clean and seperated correctly where a single commit should only change that specific part.

If looks like the CMake change are not required to make this work, if it is a pre-condition to make it work the cmake changes should have been a seperate commit before the actual code changes so we can correctly keep on using git blame and git log.

**Comment by metsma** [comment]
Fixed

File: src/pkcs11module.cpp [hunk 1]
```diff
@@ -250,27 +241,34 @@ CK_RV PKCS11Module::sign(const std::vector<unsigned char> &cert, const std::vect
...
-    case BINARY_SHA384_LENGTH:
-        hashWithPadding = {0x30, 0x41, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x02, 0x05, 0x00, 0x04, 0x30};
-        break;
-    case BINARY_SHA512_LENGTH:
-        hashWithPadding = {0x30, 0x51, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x03, 0x05, 0x00, 0x04, 0x40};
-        break;
-    default:
-        _log("incorrect digest length, dropping padding");
+    if(keyType == CKK_RSA)
+    {
```
Comment (MrJoe) [inline src/pkcs11module.cpp]: Code style in this project seems to default to 
```
if(keyType == CKK_RSA) {
```

so having the braces after the if statement. I would prefer not to mix different coding styles.

File: src/pkcs11module.cpp [hunk 2]
```diff
@@ -250,27 +241,34 @@ CK_RV PKCS11Module::sign(const std::vector<unsigned char> &cert, const std::vect
...
+        // FIXME: explicit hash type argument
```
Comment (MrJoe) [inline src/pkcs11module.cpp:252]: Does this FIXME indicate this change isn't ready yet or something else?

Comment (metsma) [reply src/pkcs11module.cpp:252]: The comment was before there already.

**Comment by MrJoe** [comment]
Thanks, indeed I missed the fact that the comment was already there!

**Pull request merged**